src/Lykke.Service.ExchangeDataStore/Lykke.Service.ExchangeDataStore.Services/DataHarvesters/OrderbookDataHarvester.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Autofac;
+using Common;
+using Common.Log;
+using Lykke.RabbitMqBroker;
+using Lykke.RabbitMqBroker.Subscriber;
+using Lykke.Service.ExchangeDataStore.Core.Domain.Events;
+using Lykke.Service.ExchangeDataStore.Core.Domain.OrderBooks;
+using Lykke.Service.ExchangeDataStore.Core.Settings.ServiceSettings;
+using Lykke.Service.ExchangeDataStore.Services.Helpers;
+using System.Threading.Tasks;
+
+// ReSharper disable MemberCanBePrivate.Global - AsyncEvent must be public
+
+namespace Lykke.Service.ExchangeDataStore.Services.DataHarvesters
+{
+    // ReSharper disable once ClassNeverInstantiated.Global - autofac instantiated
+    public class OrderbookDataHarvester : IStartable, IStopable
+    {
+        private readonly ILog _log;
+        private RabbitMqSubscriber<OrderBook> _rabbit;
+        private readonly bool _enabled;
+        private readonly RabbitMqExchangeConfiguration _orderBookQueueConfig;
+        private string Component = nameof(OrderbookDataHarvester);
+
+        public static AsyncEvent<OrderBook> OrderBookReceived = null;
+
+        protected virtual Task OnOrderBookReceived(OrderBook orderBook)
+        {
+            return OrderBookReceived.NullableInvokeAsync(this, orderBook);
+        }
+
+        public OrderbookDataHarvester(ILog log, RabbitMqExchangeConfiguration orderBookQueueConfig)
+        {
+            _log = log;
+            _enabled = orderBookQueueConfig.Enabled;
+            _orderBookQueueConfig = orderBookQueueConfig;
+            SubsribeToMessageQueue();
+        }
+
+        private void SubsribeToMessageQueue()
+        {
+            var rabbitSettings = new RabbitMqSubscriptionSettings
+            {
+                ConnectionString = _orderBookQueueConfig.ConnectionString,
+                ExchangeName = _orderBookQueueConfig.Exchange,
+                QueueName = _orderBookQueueConfig.Queue
+            };
+            var errorStrategy = new DefaultErrorHandlingStrategy(_log, rabbitSettings);
+            _rabbit = new RabbitMqSubscriber<OrderBook>(rabbitSettings, errorStrategy)
+                .SetMessageDeserializer(new GenericRabbitModelConverter<OrderBook>())
+                .SetMessageReadStrategy(new MessageReadWithTemporaryQueueStrategy())
```

Comment:
Not sure that here should be a *Temporary* queue

Reply:
Ok, I have replaced it with MessageReadQueueStrategy. Didn't find other implementations. As a note, ExchangeConnector uses MessageReadWithTemporaryQueueStrategy for all of its queues, and I am not sure why that is - shouldn't it use MessageReadQueueStrategy as well then?

Reply:
It depends on a type of the message. In your case we have already has order books from external exchanges inside Lykke and they should not be lost. With a *permanent* queue the messages will left in the queue if the service down. Other message types may have different requirements.

Reply:
You can take an example from here https://github.com/LykkeCity/Lykke.RabbitMqDotNetBroker

Reply:
**UPDATE** Changing the queue to MessageReadQueueStrategy actually stops it from receiving messages. I believe this is due to the fact that  ExchangeConnector uses MessageReadWithTemporaryQueueStrategy, so I guess if we want to use MessageReadQueueStrategy  we need to change it to MessageReadQueueStrategy   on ExchangeConnector as well. Please confirm we want to do this? Otherwise we have to leave at as it is - both using MessageReadWithTemporaryQueueStrategy, which seems to work fine.